akka-stream/src/main/scala/akka/stream/javadsl/DelayStrategy.scala

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (C) 2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.stream.javadsl
+
+import akka.stream.scaladsl
+import akka.util.JavaDurationConverters.JavaDurationOps
+
+import scala.concurrent.duration.FiniteDuration
+
+/**
+ * Allows to manage delay and can be stateful to compute delay for any sequence of elements,
+ * all elements go through nextDelay() updating state and returning delay for each element
+ */
+trait DelayStrategy[T] {
+
+  /**
+   * Returns delay for ongoing element, `Duration.Zero` means passing without delay
+   */
+  def nextDelay(elem: T): java.time.Duration
+
+}
+
+object DelayStrategy {
+
+  private[javadsl] def asScala[T](delayStrategy: DelayStrategy[T]) = new scaladsl.DelayStrategy[T] {
```

Comment:
This becomes public in Java bytecode so needs `/** Internal API */` and `@InternalApi` to make it clear that it is internal.